src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexReadOperation.cs

```diff
@@ -72,8 +79,9 @@ public CoraxIndexReadOperation(Index index, Logger logger, Transaction readTrans
             bool isBinary;
             using (coraxScope?.Start())
             {
-                if ((queryMatch = CoraxQueryBuilder.BuildQuery(_indexSearcher, null, null, query, _index, query.QueryParameters, QueryBuilderFactories, out isBinary,
-                        _fieldMappings, fieldsToFetch, highlightingTerms: highlightingTerms, take: take)) is null)
+                var queryEnv = new QueryParameters(_indexSearcher, serverContext: null, documentsContext: null, query, _index, query.QueryParameters, QueryBuilderFactories,
```

Comment:
1. `QueryParameters` sounds very generic while it's for query builder purposes only. What do you think about using `QueryBuilderParameters`?

2. `queryEnv` -> `queryParameters` or just `parameters`. Here and in all other places as well

Reply:
3. Given that instances of query parameters are created and used only in `CoraxQueryBuilder` you could move the class there. Then you could simplify the name to `Parameters` so we'd have `CoraxQueryBuilder.Parameters` and even mark is as `private` there

Reply:
done